### DISCUSSION TITLE
如何为web woker设定和publicPath不同的路径

### DISCUSSION THREAD

**Discussion by EastwoodStudio** [umijs/umi#11550] [Q&A] [answered]
webpack5 提供了内置的[worker解决方案](https://webpack.js.org/guides/web-workers/#root)
并提供了[output.workerPublicPath](https://webpack.js.org/configuration/output/#outputworkerpublicpath)来区分publicPath解决同源的问题
目前umi4的webpack版本还不够，配套的webpack-5-chain也没有升级加入新属性

有没有其他变通的方法，可以在worker加载前修改掉publicPath，事后再改回来（已经尝试了动态修改umi的dynamicPublicPath，和__webpack_public_path__, 都对URL内的地址无效）
```
  // TODO 调整publicPath
  // 启动sharedWorker
  // see https://webpack.js.org/api/module-methods/#magic-comments
  const worker = new SharedWorker(
    /* webpackChunkName: "notify-worker" */ new URL('./worker/notify/index.ts', import.meta.url),
  );

```

或许，可以升级一下 bundler-webpack ？

**Comment by fz6m** [comment, marked as answer]
这个问题要等内置 webpack 版本升级后才可以用 `workerPublicPath` 这个选项。

目前先尝试在全局注入 [`headScripts`](https://umijs.org/docs/api/config#headscripts) 等方式单独去加载这个 worker 吧，或者在代码里把路径写成绝对路径。
